tests/unit/test_dataframe.py

```diff
@@ -146,10 +146,8 @@ def test_select_bad_input():
     )
 
 
-def test_join_bad_input():
-    mock_connection = mock.create_autospec(ServerConnection)
-    mock_connection._conn = mock.MagicMock()
-    session = snowflake.snowpark.session.Session(mock_connection)
+def test_join_bad_input(mock_server_connection):
```

Comment:
Do we have a test to make sure that when multi-threading is off, only one conn is created

Reply:
let me add one.

Reply:
added in `integ/test_multithreading.py::test_num_cursors_created`